File: 스택/BOJ_10773.cpp
#include <stack>
#include <iostream>
using namespace std;
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    stack<int> S;
    int K;
    cin >> K;
    while (K--)
    {
        int num;
        cin >> num;
        if (num != 0)
            S.push(num);
        else if (num == 0)
        {
            S.pop();
        }
    }

    int sum = 0;
    for (int i = 0;; i++)
    {
        if (S.size() != 0)
        {
            sum += S.top();
            S.pop();
        }
        else
            break;
    }
    cout << sum;
}

File: 스택/BOJ_10773.java
package 스택;

import java.io.*;
import java.util.*;

public class BOJ_10773 {
    static long sum;
    static int  n,num;
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Stack<Integer> stack = new Stack<>();
        StringBuilder sb = new StringBuilder();

        n=Integer.parseInt(br.readLine());

        for(int i=0;i<n;i++){
            num=Integer.parseInt(br.readLine());
            if(num==0){
                stack.pop();
            }else{
                stack.push(num);
            }
        }

        for(Integer number:stack) sum+=number;
        System.out.print(sum);
        br.close();
    }
}
